app/services/provider_authorisation.rb

```diff
@@ -5,21 +5,25 @@ def initialize(actor:)
     @actor = actor
   end
 
-  def can_make_offer?(application_choice:)
-    @actor.is_a?(SupportUser) || @actor.providers.include?(application_choice.course.provider)
-  end
-
-  def can_change_offer?(application_choice:, course_option_id:)
-    course_option = CourseOption.find course_option_id
+  def can_make_offer?(application_choice:, course_option_id: nil)
+    supplied_course_option = CourseOption.find(course_option_id) if course_option_id
```

Comment:
Suggest we make the first if clause a return statement as this saves performing an unnecessary `find`:

```ruby
return true if @actor is_a?(SupportUser)

supplied_course_option = CourseOption.find(course_option_id) if course_option_id
if supplied_course_option && course_option_id != application_choice.course_option.id
...
```